.buildkite/scripts/dockerhub_upload.sh

```diff
@@ -4,14 +4,30 @@ set -euo pipefail
 
 source .buildkite/scripts/shared.sh
 version=$(buildkite-agent meta-data get "version")
+channel=$(buildkite-agent meta-data get "release-channel")
 
 if is_fake_release; then
   # This overrides IMAGE_NAME in docker-base.sh
   export IMAGE_NAME="habitat/fakey-mc-fake-face-studio"
+else
+  export IMAGE_NAME="habitat/default-studio"
 fi
 
+if [[ -z $DOCKER_LOGIN_USER ]]; then
+  echo "DockerHub user required" && exit 1
+fi
+
+if [[ -z $DOCKER_LOGIN_PASSWORD ]]; then
+  echo "DockerHub password required" && exit 1
+fi
+
+docker login -u="$DOCKER_LOGIN_USER" -p="$DOCKER_LOGIN_PASSWORD"
+trap 'rm -f $HOME/.docker/config.json' INT TERM EXIT
+
 pushd ./components/rootless_studio >/dev/null
-./build-docker-image.sh "${version}"
-./tag-docker-image.sh "${version}" "rc"
-./publish-to-dockerhub.sh "${version}" "rc"
+docker build -t habitat:hab-base .
+docker build --build-arg BLDR_CHANNEL="${channel}" --no-cache -t "${IMAGE_NAME}:${version}" "./default"
```

Comment:
`-t` could be `--tag`.

Also, I don't think you need to quote `"./default"`

Splitting this onto several lines could help the readability, as well.